### DISCUSSION TITLE
Constants, Static and Read-only

### DISCUSSION THREAD

**Discussion by mishelshaji** [mishelshaji/Learn-DotNet#14] [Progress Tracker]
Topics Covered:
👉 What is a constant.
👉 Declaring constants.
👉 Accessing constants inside a class.
👉 Accessing constants outside a class.
👉 When should we declare a constant.
👉 What is a static member.
👉 Declaring static members.
👉 Accessing static members inside a class.
👉 Accessing static members outside a class.
👉 When should we declare a static member.
👉 Difference between constant and static.
👉 What is a read-only member.
👉 Declaring read-only members.
👉 Accessing read-only members inside a class.
👉 Accessing read-only members outside a class.
👉 When should we declare a read-only member.
👉 Difference between constants and read-only.
👉 Updating read-only members from a constructor.
👉 Memory allocation of static and constant members.
👉 Static class.
👉 Static constructor.
👉 Why is it recommended to write a static constructor to initialize static members?

**Comment by abijithks266** [comment]
Completed

**Comment by Arjunprasad-22** [comment]
Completed

**Comment by ManikuttyBabu** [comment]
completed

**Comment by PEFathimaManfa** [comment]
completed

**Comment by Nandana-04** [comment]
Completed

**Comment by akhilamp123** [comment]
Completed

**Comment by Arun-babumon** [comment]
Completed

**Comment by AnitaMaryThomas** [comment]
Completed

**Comment by sreelakshmiChandran** [comment]
Completed

**Comment by amalapaul** [comment]
completed

**Comment by SonuSanilK** [comment]
Completed

**Comment by Jiby-Joseph-123** [comment]
Completed.

**Comment by riya45810** [comment]
Completed

**Comment by SivapriyaVB** [comment]
Completed

**Comment by jeweljose22** [comment]
Completed

**Comment by Shamla-P** [comment]
Completed

**Comment by kavitha-BS** [comment]
completed

**Comment by ARAVIND0407** [comment]
completed
